itest/rust/src/builtin_tests/script/script_instance_tests.rs

```diff
@@ -0,0 +1,181 @@
+            prop_b: false,
+            prop_list: Rc::new(vec![PropertyInfo {
+                variant_type: VariantType::Int,
+                property_name: StringName::from("script_property_a"),
+                class_name: ClassName::from_ascii_cstr("\0".as_bytes()),
+                hint: PropertyHint::PROPERTY_HINT_NONE,
+                hint_string: GString::new(),
+                usage: PropertyUsageFlags::PROPERTY_USAGE_NONE,
+            }]),
+
+            method_list: Rc::new(vec![MethodInfo {
+                id: 1,
+                method_name: StringName::from("script_method_a"),
+                class_name: ClassName::from_ascii_cstr("TestScript\0".as_bytes()),
+                return_type: PropertyInfo {
+                    variant_type: VariantType::String,
+                    class_name: ClassName::from_ascii_cstr("\0".as_bytes()),
+                    property_name: StringName::from(""),
+                    hint: PropertyHint::PROPERTY_HINT_NONE,
+                    hint_string: GString::new(),
+                    usage: PropertyUsageFlags::PROPERTY_USAGE_NONE,
+                },
+                arguments: vec![
+                    PropertyInfo {
+                        variant_type: VariantType::String,
+                        class_name: ClassName::from_ascii_cstr("\0".as_bytes()),
+                        property_name: StringName::from(""),
+                        hint: PropertyHint::PROPERTY_HINT_NONE,
+                        hint_string: GString::new(),
+                        usage: PropertyUsageFlags::PROPERTY_USAGE_NONE,
+                    },
+                    PropertyInfo {
+                        variant_type: VariantType::Int,
+                        class_name: ClassName::from_ascii_cstr("\0".as_bytes()),
+                        property_name: StringName::from(""),
+                        hint: PropertyHint::PROPERTY_HINT_NONE,
+                        hint_string: GString::new(),
+                        usage: PropertyUsageFlags::PROPERTY_USAGE_NONE,
+                    },
+                ],
+
+                default_arguments: vec![],
+                flags: MethodFlags::METHOD_FLAG_NORMAL,
+            }]),
+        }
+    }
+}
+
+impl ScriptInstance for TestScriptInstance {
+    fn class_name(&self) -> GString {
+        GString::from("TestScript")
+    }
+
+    fn set(&mut self, name: StringName, value: &Variant) -> bool {
+        match name.to_string().as_str() {
+            "script_property_b" => self.prop_b = FromGodot::from_variant(value),
+            _ => return false,
+        }
+
+        true
+    }
+
+    fn get(&self, name: StringName) -> Option<Variant> {
+        match name.to_string().as_str() {
+            "script_property_a" => Some(Variant::from(10)),
+            "script_property_b" => Some(Variant::from(self.prop_b)),
+            _ => None,
+        }
+    }
+
+    fn property_list(&self) -> Rc<Vec<PropertyInfo>> {
+        self.prop_list.clone()
+    }
```

Comment:
If this was implemented as
```rs
Rc::new(vec![/* create some new PropertyInfos here */])
```
Would we not have UB? Since the functions rely on the property infos this method returns to live for as long as `self` does.

If so then which `unsafe` block is it that is responsible for this error?

Same applies to `method_list`.

Reply:
you are right. it would be better to define this as `fn property_info(&'a self) -> &'a [PropertyInfo]`. What do you think?

Reply:
That would probably work yeah. As long as you can guarantee that `self` wont be dropped for the duration, which i think we can?

Reply:
Changed to return a slice reference.